Objects/Converters/ConverterTeklaStructures/ConverterTeklaStructuresShared/PartialClasses/ConvertDirectShapeMesh.cs

```diff
@@ -31,13 +47,24 @@ public void MeshToNative(Base @object, List<GE.Mesh> displayValues)
       // remove possibly pre-existing shape with same name and then insert a Speckle object shape
       // with that name into the catalog
       shapeItem.Delete();
+
       bool result = false;
       try
       {
         result = shapeItem.Insert();
-        // Fails if two shapes exist with different names but same geometry fingerprint
       }
-      catch (Exception ex) { }
+      catch (InvalidOperationException ex)
+      {
+        throw new ConversionException("The converted Mesh could not be inserted.", ex);
+      }
+      catch (ArgumentException ex)
+      {
+        // if the exception message contains "BRep already exists", swallow the error, else rethrow
+        if (!ex.Message.Contains("The BRep geometry already exists"))
```

Comment:
This is a bit brittle, is there a better way to identify this? Can we avoid trying the Insert() if we know it already exists?

Reply:
It could be I was being careful to maintain the current behaviour/swallow pattern but this is a good point

Reply:
The answer is no there are some internal methods for ShapeExists and no documentation of generating that `geohash` string 
<img width="539" alt="image" src="https://github.com/specklesystems/speckle-sharp/assets/760691/9db980fd-b08c-4107-aab1-911408ea4a63">

Their code examples in documentation suggest the try it and see method for `Insert()`

I am 100% open to throwing or logging and moving on for all exceptions - there are several that could be thrown:
<img width="735" alt="image" src="https://github.com/specklesystems/speckle-sharp/assets/760691/50d7337e-104d-4837-9a75-3bbaa3d865a3">

I was keeping that single exception allowable as it seemed to represent the previous comment of why the error should be swallowed.

Reply:
This is why I am keen we wrap our exceptions at some point when uses call our SDK code so that users can at least say, okay this was a SpeckleException, thing I tried didn't work.

Actually here I would catch ArgumentException, InvalidOperationException and ArgumentOutOfRangeException if that is the code we are calling into. We can then wrap it with our own context if throwing upwards - cannot see from this view.

Reply:
I started with that - but collapsed a little as `ArgumentOutOfRange` is an `ArgumentException` all we'd be doing is making a distinction where Tekla SDK could not. The pasted switch-case code is Teklas, not ours.

I am happy to do whatever we collectively want, I picked this Tekla ticket up as it has no owner :D

Reply:
All good then